fsspec/implementations/cached.py

```diff
@@ -206,25 +162,17 @@ def _check_cache(self):
     def _check_file(self, path):
         """Is path in cache and still valid"""
         path = self._strip_protocol(path)
-
         self._check_cache()
-        for storage, cache in zip(self.storage, self.cached_files):
-            if path not in cache:
-                continue
-            detail = cache[path].copy()
-            if self.check_files:
-                if detail["uid"] != self.fs.ukey(path):
-                    continue
-            if self.expiry:
-                if time.time() - detail["time"] > self.expiry:
-                    continue
-            fn = os.path.join(storage, detail["fn"])
-            if os.path.exists(fn):
-                return detail, fn
-        return False
+
+        def ignore(detail: Detail) -> bool:
```

Comment:
I am passing in this function as a callback to the `_metadata.check_file` function which needs access to information it doesn't have. There are other possible alternatives to this approach.

Reply:
I think it would be fine to have the callback signature allow/require the cache instance rather than making a dynamic function closure.

Reply:
OK, I've done that.